### PR TITLE
Added Knuth-Morris-Pratt in Javascript

### LINKED ISSUE #690: Knuth–Morris–Pratt in Javascript
This is an open issue for the Knuth–Morris–Pratt algorithm implementation in Javascript. If you wish to contribute, just make some noise and we will assign you.

### PR DESCRIPTION
**Pull request by david-azevedo**
**Fixes #690 **

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-kmp/javascript/knuth_morris_pratt.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  let j = 0;
+
+  while (i < phraseLength) {
+    
+    if (phrase[i] == word[j]) {
+      
+      i += 1;
+      j += 1;
+
+      if (j == wordLength) { // Found a match
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/javascript/knuth_morris_pratt.js:50]: Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395128)

File: algorithms/ar-kmp/javascript/knuth_morris_pratt.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+function buildPrefixTable(search) {
+  const prefixTable = Array(search.length);
+  let pos = 1
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/javascript/knuth_morris_pratt.js:3]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395128)

File: algorithms/ar-kmp/javascript/knuth_morris_pratt.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  const phraseLength = phrase.length;
+  const wordLength = word.length;
+  const prefixTable = buildPrefixTable(word);
+  const matches = [];
+  let i = 0;
+  let j = 0;
+
+  while (i < phraseLength) {
+    
+    if (phrase[i] == word[j]) {
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/javascript/knuth_morris_pratt.js:45]: Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395128)

File: algorithms/ar-kmp/javascript/knuth_morris_pratt.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
+function buildPrefixTable(search) {
+  const prefixTable = Array(search.length);
+  let pos = 1
+  let cnd = 0;
+
+  prefixTable[0] = -1;
+
+  while ( pos < search.length) {
+    
+    if (search[pos] == search[cnd]) {
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/javascript/knuth_morris_pratt.js:10]: Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395128)

File: algorithms/ar-kmp/javascript/knuth_morris_pratt.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      
+      prefixTable[pos] = prefixTable[cnd];
+    
+    } else {
+
+      prefixTable[pos] = cnd;
+      cnd = prefixTable[cnd];
+      
+      while (cnd >= 0 && search[pos] != search[cnd]) {
```
Comment (RavinduSachintha) [inline algorithms/ar-kmp/javascript/knuth_morris_pratt.js:19]: Codacy found an issue: [Expected '!==' and instead saw '!='.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395128)

**Comment by Danushka96** [comment]
Hi @david-azevedo 
Please fix the conflicts

**Comment by david-azevedo** [comment]
> Hi @david-azevedo
> Please fix the conflicts

Hey @Danushka96 ! Done! 👍

**Review by Danushka96 — APPROVED** [review]

**Pull request merged**